component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,456 @@
+import org.apache.commons.io.monitor.FileAlterationObserver;
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.log4j.Logger;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static io.siddhi.extension.io.file.util.Util.getFileHandlerEvent;
+import static io.siddhi.extension.util.Constant.STATUS_DONE;
+import static io.siddhi.extension.util.Constant.STATUS_NEW;
+import static io.siddhi.extension.util.Constant.STATUS_PROCESS;
+import static io.siddhi.extension.util.Constant.STATUS_REMOVE;
+
+
+/**
+ * Implementation of siddhi-io-file source.
+ */
+@Extension(
+        name = "fileeventlistener",
+        namespace = "source",
+        description = "" +
+                "Fileeventlistener provides the functionality for user to get the details of files " +
+                "which have been created or modified or deleted in the execution time" ,
+
+        parameters = {
+                @Parameter(
+                        name = "uri",
+                        description =
+                                "Used to specify a file to be processed. \nThe 'uri'  should be provided.\n" +
+                                        "This uri MUST have the respective protocol specified.\n",
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "monitoring.interval",
+                        description = "This parameter is used to specify the time interval (in milliseconds) " +
+                                " that the process monitor the changes for.\n",
+                        type = {DataType.STRING},
+                        optional = true,
+                        defaultValue = "1000"
+                ),
+        },
+        examples = {
+                @Example(
+                        syntax = "" +
+                                "@source(type='fileeventlistener', uri='file://abc/xyz.txt',@map(type='json')) \n" +
+                                "define stream CreateFileStream (filepath string, length long, " +
+                                "last_modified string, status string);" +
+                                "@sink(type='log')" +
+                                "define stream FooStream (filepath string, length int, " +
+                                "last_modified string, status string); \n",
+
+                        description = "" +
+                                "Under above configuration, After monitoring has been started, If the specific " +
+                                "file given in the uri gets modified or deleted. an event is created" +
```

Comment:
Can we rephrase this as 
"Under above configuration, An events is triggered if a specific file given in the URI gets modified or deleted"

Reply:
fixed in https://github.com/siddhi-io/siddhi-io-file/pull/82/commits/3f9a2a0eb50655821dd322688aa034be33ef8860